pkg/routeagent_driver/cabledriver/vxlan_tunnel_cleanup_handler.go

```diff
@@ -0,0 +1,83 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cabledriver
+
+import (
+	"fmt"
+	"syscall"
+
+	"github.com/submariner-io/submariner/pkg/event"
+
+	"github.com/submariner-io/admiral/pkg/log"
+	"github.com/submariner-io/submariner/pkg/cable/vxlan"
+	"github.com/vishvananda/netlink"
+	"k8s.io/klog"
+)
+
+type vxlanCleanup struct {
+	event.HandlerBase
+}
+
+func NewvxlanCleanup() event.Handler {
+	return &vxlanCleanup{}
+}
+
+func (h *vxlanCleanup) GetNetworkPlugins() []string {
+	return []string{event.AnyNetworkPlugin}
+}
+
+func (h *vxlanCleanup) GetName() string {
+	return "VXLAN cleanup handler"
+}
+
+func (h *vxlanCleanup) NonGatewayCleanup() error {
+	klog.V(log.DEBUG).Infof("Cleaning up the routes")
+
+	link, err := netlink.LinkByName(vxlan.VxLANIface)
+
+	if err != nil {
+		klog.Errorf("Unable to cleanup routes, error retrieving the link %s: %v", vxlan.VxLANIface, err)
+		return nil
+	}
+
+	currentRouteList, err := netlink.RouteList(link, syscall.AF_INET)
+
+	if err != nil {
+		klog.Errorf("Unable to cleanup routes, error retrieving routes on the link %s: %v", vxlan.VxLANIface, err)
```

Comment:
This should not exit at this time. Probably print the warning and not execute L65-L71, but then we really need to try delete the link interface in L73.